src/pocketmine/Player.php

```diff
@@ -2864,6 +2864,10 @@ public function handleDropItem(DropItemPacket $packet) : bool{
 			// Windows 10 Edition drops the contents of the crafting grid on container close - including air.
 			return true;
 		}
+		
+		if($this->gamemode === self::SPECTATOR) {
```

Comment:
I tested the execute times for $this->isSpectator() and $this->gamemode === self::SPECTATOR, $this->isSpectator() takes longer to execute than $this->gamemode === self::SPECTATOR. We're all about performance right? ;)